source/fundamentals/connection/tuning.txt

```diff
@@ -0,0 +1,258 @@
+==================
+Tuning Connections
+==================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. _connection-tuning-java:
+
+Overview
+--------
+
+In this guide, you can learn about **connection pools** and **connection storms**.
+
+Connection Pools
+----------------
+
+Connection pools are fundamental to the performance of your application.
+A **connection pool** is a cache of database connections maintained by the driver
+so that future requests can reuse already established connections.
+
+.. example::
+
+   Imagine you're with a group of friends at an event and make a phone
+   call to a friend that was unable to attend due to an illness. Each
+   person in the group would like to send their well-wishes. It is much
+   more efficient for you to make the phone call and then pass your phone
+   around to each friend instead of each individual making a separate
+   call one after the other.
+
+Basics
+++++++
+The driver establishes a connection pool **per connected server**. This means
+in a three member replica set, the driver maintains three connection pools,
+one for each member of the replica set. In a sharded deployment, the driver
+maintains one connection per ``mongos``.
+
+In MongoDB drivers, there are 4 key connection pool options:
+
+- ``minPoolSize``: The minimum number of connections that may be
+  associated with a pool at a given time, including in use and available
+  connections. The default is ``0``.
+
+- ``maxPoolSize``: The maximum number of connections that may be
+  associated with a pool at a given time, including in use and available
+  connections. The default is ``100``.
+
+- ``maxIdleTimeMS``: The maximum number of milliseconds that a connection
+  associated with a pool may be idle before being removed from the pool.
+  The default is ``0``, which indicates that idle connections are not
+  removed.
+
+- ``maxConnecting``: The maximum number of connections to each server that
```

Comment:
Rather than phrasing this as a number of connections that the driver _will_ establish at once, I think it would be better to say its the maximum that it _can_ (i.e. a limit).

e.g.
"The maximum number of connections that the driver can concurrently establish to an individual server.", or something like that.